CRM/Contribute/BAO/Query.php

```diff
@@ -386,21 +246,14 @@ public static function whereClauseSingle(&$values, &$query) {
       case (strpos($name, '_amount') !== FALSE):
       case (strpos($name, '_date') !== FALSE && $name != 'contribution_fulfilled_date'):
       case 'contribution_campaign_id':
-        $qillName = $name;
-        $pseudoExtraParam = array();
-        // @todo including names using a switch statement & then using an 'if' to filter them out is ... odd!
-        if ((strpos($name, '_amount') !== FALSE) || (strpos($name, '_date') !== FALSE) || in_array($name,
-            array(
-              'contribution_id',
-              'contribution_currency',
-              'contribution_source',
-              'contribution_trxn_id',
-              'contribution_check_number',
-              'contribution_payment_instrument_id',
-              'contribution_contact_id',
-              'contribution_campaign_id',
-            )
-          )
+
+        $fieldNamesNotToStripContributionFrom = array(
```

Comment:
Note we could use 'presence in $this->fields' as a metadata based way to opt in or out